packages/react/src/generators/federate-module/federate-module.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  GeneratorCallback,
+  Tree,
+  convertNxGenerator,
+  formatFiles,
+  logger,
+  readJson,
+  runTasksInSerial,
+} from '@nx/devkit';
+import { Schema } from './schema';
+
+import { remoteGeneratorInternal } from '../remote/remote';
+import { addPathToExposes, checkRemoteExists } from './lib/utils';
+import { determineProjectNameAndRootOptions } from '@nx/devkit/src/generators/project-name-and-root-utils';
+import { addTsConfigPath, getRootTsConfigPathInTree } from '@nx/js';
+
+export async function federateModuleGenerator(tree: Tree, schema: Schema) {
+  const tasks: GeneratorCallback[] = [];
+  // Check remote exists
+  const remote = checkRemoteExists(tree, schema.remote);
+  const { projectName, projectRoot: remoteRoot } =
+    await determineProjectNameAndRootOptions(tree, {
+      name: schema.remote,
+      projectType: 'application',
+      projectNameAndRootFormat: schema.projectNameAndRootFormat,
+      callingGenerator: '@nx/react:federate-module',
+    });
+
+  if (!remote) {
+    // create remote
+    const remoteGenerator = await remoteGeneratorInternal(tree, {
+      name: schema.remote,
+      e2eTestRunner: schema.e2eTestRunner,
+      skipFormat: schema.skipFormat,
+      linter: schema.linter,
+      style: schema.style,
+      unitTestRunner: schema.unitTestRunner,
+      host: schema.host,
+      projectNameAndRootFormat: schema.projectNameAndRootFormat ?? 'derived',
+    });
+
+    tasks.push(remoteGenerator);
+  }
+
+  const projectRoot = remote ? remote.root : remoteRoot;
+  const remoteName = remote ? remote.name : projectName;
+
+  // add path to exposes property
+  addPathToExposes(tree, projectRoot, schema.name, schema.path);
```

Comment:
just want to double check, remote could be optional, but path should be required in the schema, right?

Reply:
They are both required. I added them to `required: []` property inside `schema.json`